packages/application-components/src/components/tab-header/tab-header.spec.tsx

```diff
@@ -0,0 +1,51 @@
+import { warning } from '@commercetools-uikit/utils';
+import { screen, renderComponent, fireEvent, waitFor } from '../../test-utils';
+import TabHeader from './tab-header';
+
+jest.mock('@commercetools-uikit/utils', () => ({
+  ...jest.requireActual('@commercetools-uikit/utils'),
+  warning: jest.fn(),
+}));
+
+const renderTabHeader = (additionalProps = {}) =>
+  renderComponent(
+    <TabHeader to="/tab-one" label="Tab One" {...additionalProps} />
+  );
+
+describe('rendering', () => {
+  it('should render when "to" is string', () => {
+    renderTabHeader();
+
+    screen.getByText(/tab one/i);
+  });
+  it('should render when "to" is of LocationProvider type', () => {
+    renderTabHeader({ to: { pathname: '/tab-one' } });
+
+    screen.getByText(/tab one/i);
+  });
+  it('should display a custom intl message instead of a label', () => {
+    const intlMessage = { id: 'localized', defaultMessage: 'localized text' };
+    renderTabHeader({ intlMessage });
```

Comment:
In this case the `label` prop should not be passed.

Reply:
How about leaving it to show that `intlMessage` takes precedence over `label` when both are passed and adding another test with `intlMessage` and without passing the `label` prop?

Reply:
Sure, that works too. Thanks